geoh5py/shared/entity_type.py

```diff
@@ -20,136 +20,209 @@
 import uuid
 import weakref
 from abc import ABC, abstractmethod
-from typing import TYPE_CHECKING, TypeVar, cast
+from typing import TYPE_CHECKING, Any, TypeVar, cast
 
 if TYPE_CHECKING:
-    from .. import workspace as ws
-
+    from ..workspace import Workspace
 
 EntityTypeT = TypeVar("EntityTypeT", bound="EntityType")
 
 
 class EntityType(ABC):
-    _attribute_map = {"Description": "description", "ID": "uid", "Name": "name"}
+    """
+    The base class for all entity types.
 
-    def __init__(self, workspace: ws.Workspace, uid: uuid.UUID | None = None, **kwargs):
-        assert workspace is not None
-        self._workspace = weakref.ref(workspace)
+    :param workspace: The workspace to which the entity type belongs.
+    :param uid: The unique identifier of the entity type.
+    :param description: The description of the entity type.
+    :param name: The name of the entity type.
+    :param kwargs: Additional keyword arguments to set as attributes.
+    """
 
-        assert uid is None or isinstance(uid, uuid.UUID)
-        self._description: str | None = "Entity"
-        self._name: str | None = "Entity"
-        self._on_file = False
-        self._uid: uuid.UUID = uid if uid is not None else uuid.uuid4()
```

Comment:
No, we need to define the private attributes.

Reply:
pylint/mypy doesn't dislike setting init with setter. 
It get the logic. 
Are you sure about this practice?